api/v1/controllers/collectors.js

```diff
@@ -458,14 +489,28 @@ function stopCollector(req, res, next) {
   req.swagger.params.queryBody = {
     value: { status: 'Stopped' },
   };
-  doPatch(req, res, next, helper);
+
```

Comment:
I didn't realize we would have to duplicate the code from doPatch here in order to make this work.
I'm wondering if it might be better to call reassignGenerators in db hooks instead?

Reply:
So, in collector afterUpdate hook, we check if the status is changed to 'Paused' or 'Stopped', then we call reassignGenerators? Sounds good to me. I don't see any disadvantage.

Reply:
yeah.

Reply:
And we could also check if it was changed to 'Started', and move the assignment logic from above.